tests/UnifiedSpecTests/Util.php

```diff
@@ -26,13 +34,113 @@
+            'insertMany' => ['options', 'documents', 'session', 'ordered', 'bypassDocumentValidation'],
+            'insertOne' => ['document', 'session', 'bypassDocumentValidation'],
+            'listIndexes' => ['session', 'maxTimeMS'],
+            'mapReduce' => ['map', 'reduce', 'out', 'session', 'typeMap', 'bypassDocumentValidation', 'collation', 'finalize', 'jsMode', 'limit', 'maxTimeMS', 'query', 'scope', 'sort', 'verbose'],
+        ],
+        ChangeStream::class => [
+            'iterateUntilDocumentOrError' => [],
+        ],
+        Cursor::class => [
+            'close' => [],
+            'iterateUntilDocumentOrError' => [],
+        ],
+        Session::class => [
+            'abortTransaction' => [],
+            'commitTransaction' => [],
+            'endSession' => [],
+            'startTransaction' => [],
+            'withTransaction' => ['callback', 'readConcern', 'writeConcern'],
+        ],
+        Bucket::class => [
+            'delete' => ['id'],
+            'downloadByName' => ['filename', 'revision'],
+            'download' => ['id'],
+            'uploadWithId' => ['id', 'filename', 'source', 'chunkSizeBytes', 'disableMD5', 'contentType', 'metadata'],
+            'upload' => ['filename', 'source', 'chunkSizeBytes', 'disableMD5', 'contentType', 'metadata'],
+        ],
+    ];
+
     public static function assertHasOnlyKeys($arrayOrObject, array $keys): void
     {
         assertThat($arrayOrObject, logicalOr(isType('array'), isInstanceOf(stdClass::class)));
         $diff = array_diff_key((array) $arrayOrObject, array_fill_keys($keys, 1));
         assertEmpty($diff, 'Unsupported keys: ' . implode(',', array_keys($diff)));
     }
 
+    public static function assertArgumentsBySchema(string $executingObjectName, string $operation, array $args): void
+    {
+        assertArrayHasKey($executingObjectName, self::$args, 'Missing object "' . $executingObjectName . '" definition. The list of allowed objects, operation names and arguments should be defined at ' . self::class);
+        assertIsArray(self::$args[$executingObjectName], 'Object "' . $executingObjectName . '" is not correctly defined. The list of allowed objects, operation names and arguments should be defined at ' . self::class . '. ');
+        assertArrayHasKey($operation, self::$args[$executingObjectName], 'Missing operation "' . $operation . '" definition for object "' . $executingObjectName . '". The list of allowed objects, operation names and arguments should be defined at ' . self::class);
+        assertIsArray(self::$args[$executingObjectName][$operation], 'Operation "' . $operation . '" for object "' . $executingObjectName . '" is not correctly defined. The list of allowed objects, operation names and arguments should be defined at ' . self::class);
+        self::assertHasOnlyKeys($args, self::$args[$executingObjectName][$operation]);
```

Comment:
```suggestion
        self::assertHasOnlyKeys($args, self::$args[$executingObjectName][$operation]);
```

Thinking about this some more in light of my [original comment](https://github.com/mongodb/mongo-php-library/pull/898#discussion_r824764788), I'd prefer to use a more concise approach.

`assertArgumentsBySchema()` is only going to be called from validate operation cases, so there really should not be a situation where `self::$args[$executingObjectName][$operation]` is not set. Nevertheless, we can avoid using the null-coalescing operator (i.e. `??`) as I originally proposed and invite a PHP error. That will provide the same sanity check without the overhead of  four additional assertions per operation.